src/components/Table/utils/RowStates.ts

```diff
@@ -43,37 +49,32 @@ export class RowStates {
     const states = this;
     const map = this.map;
 
-    function addRowAndChildren(row: GridDataRow<any>): RowState {
+    function addRowAndChildren(parent: RowState | undefined, row: GridDataRow<any>): RowState {
       // This should really be kind+id, but a lot of our lookups just use id
       const key = row.id;
       let state = map.get(key);
       if (!state) {
-        state = new RowState(states, row);
+        state = new RowState(states, parent, row);
         map.set(key, state);
       } else {
+        state.parent = parent;
         state.row = row;
         state.removed = false;
         existing.delete(state);
       }
```

Comment:
Good questions! For 1, mobx will basically do that for us, if we do `foo.state = "bar"` and the value was already "bar" (or whatever instance of `parent`), it will not register as a change, and so will not trigger downstream reactions.

For 2, we remove it from existing to basically mark it has claimed, so at the end, anything left in existing, i.e. unclaimed, qualifies as a removed row, i.e. will turn into a kept row.

Maybe the "claimed" terminology would be better?